builder/builder.go

```diff
@@ -2,19 +2,48 @@ package builder
 
 import (
 	"archive/tar"
-	"assignment-exec/image-builder/constants"
+	"assignment-exec/image-builder/environment"
 	"bytes"
 	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
 	"io"
 	"io/ioutil"
 	"log"
 	"os"
 )
 
+type dockerAuthData struct {
+	Username   string `yaml:"username"`
+	Password   string `yaml:"password"`
+	Repository string `yaml:"repository"`
+	Version    string `yaml:"version"`
+}
+
+// Get the docker authentication details.
+func GetAuthData(filename string) (*dockerAuthData, error) {
+
+	yamlFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read log config file")
+	}
+
+	c := &dockerAuthData{}
+	err = yaml.Unmarshal(yamlFile, c)
+	if err != nil {
+		log.Fatalf("error in unmarshalling yaml: %v", err)
+	}
+
+	return c, nil
+}
+
 // Builds Docker Image for Code Runner.
```

Comment:
Maybe just have the comment has "Build Docker Image".

Reply:
Okay